### PR TITLE
Event functions

### PR DESCRIPTION
**Pull request by AlexanderPrendota**

### CODE REVIEW

File: examples.md [hunk 1]
```diff
@@ -279,7 +279,7 @@ Look at example:
 </div>
 ```
 
-<button onclick="KotlinPlayground('.kotlin-code-2'); this.disabled = true; document.getElementById('kotlin-example').style.display = 'block';">Create</button>
+<button onclick="KotlinPlayground('.kotlin-code-2',{ onChange: (code)=> {console.log(code)}}); this.disabled = true; document.getElementById('kotlin-example').style.display = 'block';">Create</button>
```
Comment (EvgeniiRay) [inline examples.md]: console.log?

File: examples/index.html [hunk 2]
```diff
@@ -222,15 +222,15 @@ <h2>Manual initialization</h2>
   <span class="hljs-tag">&lt;/<span class="hljs-name">textarea</span>&gt;</span>
 <span class="hljs-tag">&lt;/<span class="hljs-name">div</span>&gt;</span>
 </code></pre>
-<p><button onclick="KotlinPlayground('.kotlin-code-2'); this.disabled = true; document.getElementById('kotlin-example').style.display = 'block';">Create</button></p>
+<p><button onclick="KotlinPlayground('.kotlin-code-2',{ onChange: (code)=> {console.log(code)}}); this.disabled = true; document.getElementById('kotlin-example').style.display = 'block';">Create</button></p>
```
Comment (EvgeniiRay) [inline examples/index.html]: console.log?

File: src/index.js [hunk 3]
```diff
@@ -6,11 +6,18 @@ import {
 } from './discourse-preview-panel-handler';
 
 /**
+ *
+ * @typedef {Object} eventFunctions
+ * @property {Function} onChange
+ * @property {Function} onTestPassed
+ * @property {Function} callBack
```
Comment (EvgeniiRay) [inline src/index.js]: callback * no capital letters

File: src/view/output-view.js [hunk 4]
```diff
@@ -11,9 +11,9 @@ const ANGLE_BRACKETS_LEFT_HTML = "&lt;";
 const ANGLE_BRACKETS_RIGHT_HTML = "&gt;";
 
 const TEST_STATUS = {
-  FAIL: 'Fail',
-  ERROR: 'Error',
-  PASSED: 'Passed'
+  FAIL : { value: "FAIL", text: "Fail" },
+  ERROR: { value: "ERROR", text: "Error" },
```
Comment (EvgeniiRay) [inline src/view/output-view.js]: а че кавычки везде разные )))))))))))))))))) (с)

File: src/view/output-view.js [hunk 5]
```diff
@@ -32,24 +32,27 @@ export function processJVMOutput(output, theme) {
...
 
-export function processJUnitResults(data) {
+export function processJUnitResults(data, onTestPassed) {
   let result = "";
   let totalTime = 0;
+  let passed = true;
   if (isEmptyObject(data)) return NO_TEST_FOUND;
   for (let testClass in data) {
     let listOfResults = arrayFrom(data[testClass]);
     result = result + listOfResults.reduce((previousTest, currentTest) => {
```
Comment (EvgeniiRay) [inline src/view/output-view.js]: result+=

File: src/view/output-view.js [hunk 6]
```diff
@@ -32,24 +32,27 @@ export function processJVMOutput(output, theme) {
...
       totalTime = totalTime + (currentTest.executionTime / 1000);
+      if (currentTest.status === TEST_STATUS.ERROR.value || currentTest.status === TEST_STATUS.FAIL.value) passed = false;
       switch (currentTest.status) {
-        case "FAIL":
-          return previousTest + `<span class="console-icon fail"></span><div class="test-fail">${TEST_STATUS.FAIL}: ${currentTest.methodName}: ${convertToHtmlTag(currentTest.comparisonFailure.message)}</div>`;
-        case "ERROR":
-          return previousTest + `<span class="console-icon fail"></span><div class="test-fail">${TEST_STATUS.ERROR}: ${currentTest.methodName}: ${convertToHtmlTag(currentTest.exception.message)}</div>`;
-        case "OK":
-          return previousTest + `<span class="console-icon ok"></span><div class="test-output">${TEST_STATUS.PASSED}: ${currentTest.methodName}</div>`;
+        case TEST_STATUS.FAIL.value:
```
Comment (EvgeniiRay) [inline src/view/output-view.js]: why don't you use just enum without value prefix

File: examples/index.html (hunk 2 above)
Comment (AlexanderPrendota) [reply examples/index.html]: console.log

File: examples.md (hunk 1 above)
Comment (AlexanderPrendota) [reply examples.md]: console.log

File: src/view/output-view.js (hunk 6 above)
Comment (AlexanderPrendota) [reply src/view/output-view.js]: because it's not a ts

**Pull request merged**